src/xtgeo/xyz/polygons.py

```diff
@@ -93,26 +259,105 @@ class Polygons(XYZ):  # pylint: disable=too-many-public-methods
-        """Polygons() initialisation."""
-        self._hname = "H_CUMLEN"
-        self._dhname = "H_DELTALEN"
-        self._tname = "T_CUMLEN"
-        self._dtname = "T_DELTALEN"
-        self._name = "poly"  # the name of the Polygons() instance
-        super().__init__(*args, **kwargs)
+    @_allow_deprecated_init
+    def __init__(
+        self,
+        values: Union[list, np.ndarray, pd.DataFrame] = None,
+        xname: str = "X_UTME",
+        yname: str = "Y_UTMN",
+        zname: str = "Z_TVDSS",
+        pname: str = "POLY_ID",
+        hname: str = "H_CUMLEN",
+        dhname: str = "H_DELTALEN",
+        tname: str = "T_CUMLEN",
+        dtname: str = "T_DELTALEN",
+        name: str = "poly",
+        attributes: Optional[dict] = None,
+    ):
+        super().__init__(xname, yname, zname)
+
+        if values is None:
+            values = []
+
+        self._reset(
+            values=values,
+            xname=xname,
+            yname=yname,
+            zname=zname,
+            pname=pname,
+            hname=hname,
+            dhname=dhname,
+            tname=tname,
+            dtname=dtname,
+            name=name,
+            attributes=attributes,
+        )
 
-        self._ispolygons = True
+    def _reset(
+        self,
+        values: Union[list, np.ndarray, pd.DataFrame],
+        xname: str = "X_UTME",
+        yname: str = "Y_UTMN",
+        zname: str = "Z_TVDSS",
+        pname: str = "POLY_ID",
+        hname: str = "H_CUMLEN",
+        dhname: str = "H_DELTALEN",
+        tname: str = "T_CUMLEN",
+        dtname: str = "T_DELTALEN",
+        name: str = "poly",
+        attributes: Optional[dict] = None,
+    ):
+        """Used in deprecated methods."""
 
-    def __str__(self):
-        """User friendly print."""
-        return self.describe(flush=False)
+        super()._reset(xname, yname, zname)
+        # additonal state properties for Polygons
+        self._pname = pname
 
-    # ----------------------------------------------------------------------------------
-    # Properties
-    # ----------------------------------------------------------------------------------
+        self._hname = hname
+        self._dhname = dhname
+        self._tname = tname
+        self._dtname = dtname
+        self._name = name
+
+        if not isinstance(values, pd.DataFrame):
+            self._df = _xyz_io._from_list_like(values, self._zname, attributes, True)
+        else:
+            self._df = values
+
+    def __repr__(self):
+        return (
+            f"{self.__class__.__name__}("
+            f"values={repr(self._df)}, "
+            f"xname={self.xname}, "
+            f"yname={self.yname}, "
+            f"zname={self.zname}, "
+            f"pname={self.pname}, "
+            f"hname={self.hname}, "
+            f"dhname={self.dhname}, "
+            f"tname={self.tname}, "
+            f"dtname={self.dtname}, "
+            f"name={self.name}, "
+        )
```

Comment:
Same as Points; why changed?

Reply:
It was technically just missing a `__repr__`.